tests/definition_setup.py

```diff
@@ -72,42 +72,64 @@ def make_assign(
     )
 
 
-def make_definition(
-    name: str, fields: List[str], domain: Domain, body: BodyType, iteration_order: IterationOrder
+def make_definition_multiple(
+    name: str,
+    fields: List[str],
+    domain: Domain,
+    info: List[Tuple[BodyType, IterationOrder, Tuple[int, int]]],
 ) -> StencilDefinition:
     api_signature = [ArgumentInfo(name=n, is_keyword=False) for n in fields]
-    tmp_fields = {i[0] for i in body}.union({i[1] for i in body}).difference(fields)
+    bodies = []
+    for definition in info:
+        bodies.extend(definition[0])
+    tmp_fields = {i[0] for i in bodies}.union({i[1] for i in bodies}).difference(fields)
```

Comment:
Are you sure this does the same thing as before?
it seems like the expression `{i[X] for i in bodies} will now create a set of `BodyType` rather than field names...

Reply:
Ah, I think I miscounted nesting levels. It has become a little hard to follow compared to before

Reply:
I have some ideas, I will get back to you when I have tried them.

Reply:
@jdahm I made a pull request into this branch on your fork that generalizes this setup in a more extensible way.

If you like it, you can simply merge it, else I hope you can build on it.

Reply:
Merged! Thanks again.